xpod.go

```diff
@@ -8,99 +9,146 @@ import (
+	}
 
-	// Find shows
+	// Find episodes
 	doc.Find("div.broadcast.cfm-has-audio > div.info > div.title > a").Each(func(i int, s *goquery.Selection) {
 		// Find the link and title
-		link, _ := s.Attr("href")
-		item, _ := make_item(t_site + link)
-		feed.Items = append(feed.Items, &item)
+		link, exists := s.Attr("href")
+		if !exists {
+			log.Printf("Can't find episode href for `%s'", url)
+			return // Skip
+		}
+		item, err := make_item(t_site + link)
+		if err != nil {
+			log.Printf("Couldn't make Item for `%s': %s", url, err)
+			return // Skip
+		}
+		feed.Items = append(feed.Items, item)
 	})
-	return feed, nil
+
+	return &feed, nil
 }
 
-func make_item(url string) (feeds.Item, error) {
-	var itm feeds.Item
+// Parse a timestamp string into a time.Time
+// If parsing fails, return the zero-value timestamp.
+func parse_time(time_s string) (time.Time, error) {
+	var ts time.Time
+
+	ts, err := time.Parse("3:00pm, 1-2-2006", time_s)
+	if err != nil {
+		return ts, errors.Errorf("Failed to parse timestamp `%s'", time_s)
+	}
+
+	return ts, nil
+}
 
-	doc, err := fetch(url)
+// Create a feed Item from the URL to an episode of a show.
+func make_item(url string) (*feeds.Item, error) {
+	resp, doc, err := fetch(url)
 	if err != nil {
-		return itm, errors.Wrapf(err, "Failed to get episode page `%s'", url)
+		return nil, errors.Wrapf(err, "Failed to get episode page `%s'", url)
 	}
 
 	node := doc.Find("a.player")
-	enclosure_url, _ := node.Attr("href")
+	enclosure_link, exists := node.Attr("href")
+	if !exists {
+		return nil, errors.New("Couldn't find a.player href attr")
+	}
 
-	itm.Title = node.Text()
-	itm.Link = &feeds.Link{Href: url, Rel: "canonical"}
-	itm.Enclosure = &feeds.Enclosure{Url: enclosure_url}
-	itm.Content = doc.Find("div.creek-playlist").Text()
+	enclosure_url := enclosure_link
+	parsed_url, err := resp.Request.URL.Parse(enclosure_link)
+	if err != nil {
+		enclosure_url = parsed_url.String()
+	}
 
-	return itm, nil
+	created, err := parse_time(doc.Find("div.date").Text())
+	if err != nil {
+		log.Printf("Falling back to nil timestamp: %s", err)
+	}
+
+	itm := feeds.Item{
+		Title:   node.Text(),
+		Created: created,
+		Link:    &feeds.Link{Href: url, Rel: "canonical"},
+		Enclosure: &feeds.Enclosure{
+			Url:    enclosure_url,
+			Length: "unknown",
+			Type:   "audio/mpeg",
+		},
+		Content: doc.Find("div.creek-playlist").Text(),
+	}
+
+	return &itm, nil
 }
 
 func main() {
-	shows := []string{"heavy-metal-sewing-circle", "gothique-boutique", "sfutf"}
+	exit_status := 0
+	verbose := true
 
-	for _, show := range shows {
+	for _, show := range os.Args[1:] {
 		url := t_shows + "/" + show
-		feed_file := show + ".xml"
-		fmt.Printf("%s -> %s\n", url, feed_file)
+		feed_file := show + ".rss"
+		if verbose {
+			fmt.Printf("%s -> %s\n", url, feed_file)
+		}
 
 		f, err := os.Create(feed_file)
 		if err != nil {
 			log.Printf("Failed to open `%s': %s", feed_file, err)
+			exit_status++
 			continue
 		}
```

Comment:
```suggestion
		}
		defer f.Close()
```

You should `defer` the `Close()` to make sure the file gets closed even if the next chunk of code `continue`s.

(I mean, you're in `main` so it's gonna all get cleaned up at the end anyway, but still...)